flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -829,7 +833,8 @@ object AggregateUtil {
       outputType: RelDataType,
       groupings: Array[Int]): (Option[DataSetPreAggFunction],
         Option[TypeInformation[Row]],
-        RichGroupReduceFunction[Row, Row]) = {
+        RichGroupReduceFunction[Row, Row],
+        Option[MapPartitionFunction[Row, Row]]) = {
```

Comment:
Please remove this parameter and extend `DataSetFinalAggFunction` to also implement `RichMapPartitionFunction` such that it can be used as both (similar to `DataSetPreAggFunction`).